tests/lib/rules/use-object-in-propequal.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview Enforce use of objects as expected values in `assert.propEqual`
+ * @author Kevin Partington
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/use-object-in-propequal"),
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+function wrap(assertionCode, testName) {
+    testName = testName || "Name";
+    return `QUnit.test('${testName}', function (assert) { ${assertionCode} });`;
+}
+
+function createInvalid(assertionCode, invalidValue) {
+    return {
+        code: wrap(assertionCode),
+        errors: [{
+            messageId: "useObject",
+            data: {
+                value: invalidValue
+            }
+        }]
+    };
+}
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+    parserOptions: {
+        ecmaVersion: 2015,
+        ecmaFeatures: { jsx: true }
+    }
+});
+
+ruleTester.run("use-object-in-propequal", rule, {
+    valid: [
```

Comment:
A few more valid:

```js
wrap("assert.propEqual(actual, foo?.bar);"),
wrap("assert.propEqual(actual, foo?.bar?.());"),
```

Reply:
I'm not sure I can write tests for optional chaining as I think those are in ES2020, which ESLint 5 doesn't support. (Similar to the commented or bigint case.) I can add them commented or. Let me know if I'm mistaken about when optional chaining was introduced.